### PR TITLE
Implement lyrics encoder

### PR DESCRIPTION
**Pull request by minsing-jin**
Implement lyrics encoder using perplexity. I will check this later.

image

### CODE REVIEW

**Comment by minsing-jin** [comment]
Finally, I found that the Glow-TTS implementation was exactly what I was looking for—haha. A "stacked encoder" means that the output of one encoder is used as the input to another encoder, not the summation of all encoder's outputs.

So, I implemented an FFTEncoder for both LyricsEncoder and EnhancedEncoder. Now, I'd like to discuss whether it makes sense to separate LyricsEncoder and EnhancedConditionEncoder, since they both share the same FFTEncoder functionality.

**Comment by vkehfdl1** [comment]
@minsing-jin Each encoder shares same model, so don't need to make new modeling. But I think we need to make a separate DataLoader and train files to each encoders, because it consumes different data for embedding and training (contrastive loss)

**Comment by minsing-jin** [comment]
> @minsing-jin Each encoder shares same model, so don't need to make new modeling. But I think we need to make a separate DataLoader and train files to each encoders, because it consumes different data for embedding and training (contrastive loss)

I agree. I think we need to implement an eventuality-conditional encoder class using the FFT encoder, melody-U,  and timbre.

**Comment by minsing-jin** [comment]
image

**Review by vkehfdl1 — APPROVED** [review]
LGTM

**Review by vkehfdl1 — APPROVED** [review]
LGTM

File: songsuro/modules/fft/attentions.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
+import math
+import torch
+from torch import nn
+from torch.nn import functional as F
+
+from songsuro.modules.fft.layer_norm import LayerNorm
+
+
+class Encoder(nn.Module):
```
[the remark this answers is not in the document]
Comment (minsing-jin) [reply songsuro/modules/fft/attentions.py:9]: Lodongguen

**Review by bwook00 — APPROVED** [review]
LGTM No more Review!

**Review by vkehfdl1 — APPROVED** [review]
LGTM
